public/app/plugins/datasource/prometheus/querybuilder/shared/QueryBuilderHints.tsx

```diff
@@ -36,15 +37,25 @@ export const QueryBuilderHints = <T extends PromLokiVisualQuery>({
     setHints(hints);
   }, [datasource, visualQuery, data, queryModeller]);
 
+  const hasLabelFilter = visualQuery.operations.find((x) => x.id === '__label_filter');
+
   return (
     <>
       {hints.length > 0 && (
         <div className={styles.container}>
           {hints.map((hint) => {
+            if (hint.type === 'ADD_LABEL_FILTER' && hasLabelFilter) {
+              return null;
+            }
```

Comment:
Instead of this, I'd rather remove "hiding" of empty label filter that we do here https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/operationUtils.ts#L151-L153. 

`labelFilterRenderer` is a function that determines how is label filter operation in query builder going to be shown as query text. And what we do now is that we don't show anything if first parameter is empty string. But in this case, I'd consider removing this logic and just showing incomplete label filter.

Reply:
i didn;t know this existed, thank you!

Reply:
I recommend checking out https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/operations.ts to see the structure of query builder operations :)